lib/ui/sign_transaction/interactor/sign_transaction_bloc.dart

```diff
@@ -33,9 +37,20 @@ class SignTransactionBloc extends Bloc<SignTransactionEvent, SignTransactionStat
   Future<void> _initial(_Initial event, Emitter<SignTransactionState> emit) async {}
 
   FutureOr<void> _onUserSlideCompleted(_OnUserSlideCompleted event, Emitter<SignTransactionState> emit) async {
+    /// Try to create a free transaction
+
+    /// Would be a lot nicer if we can get the authenticated user somewhere from the bloc
+    /// Would also work better for supporting multiple accounts.
+    ///
+    /// This is also the wrong place to try to create a free transaction, it needs to happen upstream when we show the data on
+    /// the transaction screen - this code here only gets executed when user signs and slides right
+    /// But the transaction transformation is now tested, and the execution is also tested.
+    final authenticated = GetIt.I.get<AuthRepository>().authDataOrCrash;

```

Comment:
inject the auth repo into the bloc.

Reply:
yeah well the entire code shouldn't be there... fixing this